### ISSUE TITLE
UploaderExtension trigger E_USER_DEPRECATED

### ISSUE THREAD

**Issue by laurentmuller** [dustin10/VichUploaderBundle#1301]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 1.20.0

#### Summary

When using the `UploaderExtension` class an E_USER_DEPRECATED is raised. 

#### Previous behavior

No E_USER_DEPRECATED was raised.

#### Current behavior

When using the  `vich_uploader_asset` filter the follwing error is raised:

```
User Deprecated: Passing a classname is deprecated and will be removed in version 2.
```
This is because the `$className` parameter is used in the function. We have no possibility to skip this argument.

#### How to reproduce

Simply use the `vich_uploader_asset` filter in a Twig template.

#### How to resolve

Just remove the third parameter (`$className`).

**Comment by garak** [comment]
Who is raising deprecation?

**Comment by endelwar** [comment]
This line does:
https://github.com/dustin10/VichUploaderBundle/blob/6b8ac89b54ee8340e3fc108927b2c20226118436/src/Twig/Extension/UploaderExtension.php#L57

For fixing this I would not remove the third parameter, because user doesn't get warned of deprecation: instead I would like a solution like this:

```php
if ($className === null) {
	return $this->helper->asset($object, $fieldName);
}

return $this->helper->asset($object, $fieldName, $className);
```

**Comment by endelwar** [comment]
Version 2 of this bundle: see https://github.com/dustin10/VichUploaderBundle/pull/1288 that introduced this deprecation warning

**Comment by garak** [comment]
Sorry, I didn't recall that.
So, it looks like deprecation is needed.
Moreover, I don't see how adding a deprecation in a minor version could be considered a BC break.

**Comment by endelwar** [comment]
> So, it looks like deprecation is needed.
> Moreover, I don't see how adding a deprecation in a minor version could be considered a BC break.

I agree with you, it's not BC break, but IMHO internal code should not call deprecated features.

This twig fragment triggers deprecation, but it should not, I'm not passing `$className`, internal bundle code is:
```twig
<img src="{{ vich_uploader_asset(product) }}">
```

**Comment by garak** [comment]
You're right.
I agree with the proposed solution. Please open a PR.

**Comment by endelwar** [comment]
Any chances for a 1.20.1 release including this fix?

**Comment by garak** [comment]
Released directly 1.21
